sycl/test/basic_tests/image_accessor_readsampler.cpp

```diff
@@ -192,12 +204,199 @@ void checkSamplerNearest() {
+  // addressing_mode::clamp_to_edge
+  {
+    s::cl_float4 Coord(0.0f, 1.5f, 2.5f, 0.0f); // Out-of-range
+    auto Sampler = s::sampler(s::coordinate_normalization_mode::unnormalized,
+                              s::addressing_mode::clamp_to_edge,
+                              s::filtering_mode::linear);
+    checkReadSampler<6>(host_ptr, Sampler, Coord,
+                        s::cl_float4((56.0f / 127.0f), (57.0f / 127.0f),
+                                     (58.0f / 127.0f),
+                                     (59.0f / 127.0f)) /*Expected Value*/,
+                        15000 /*Precision in ULP*/);
+  }
+  {
+    s::cl_float4 Coord(0.0f, 0.2f, 0.5f, 0.0f); // In-range
+    auto Sampler = s::sampler(s::coordinate_normalization_mode::unnormalized,
+                              s::addressing_mode::clamp_to_edge,
+                              s::filtering_mode::linear);
+    checkReadSampler<6>(host_ptr, Sampler, Coord,
+                        s::cl_float4((0.0f / 127.0f), (1.0f / 127.0f),
+                                     (2.0f / 127.0f),
+                                     (3.0f / 127.0f)) /*Expected Value*/,
+                        15000 /*Precision in ULP*/);
+  }
+
+  // addressing_mode::clamp
+  {
+    s::cl_float4 Coord(0.0f, 1.5f, 1.5f, 0.0f); // Out-of-range
+    auto Sampler =
+        s::sampler(s::coordinate_normalization_mode::unnormalized,
+                   s::addressing_mode::clamp, s::filtering_mode::linear);
+    checkReadSampler<7>(host_ptr, Sampler, Coord,
+                        s::cl_float4((16.0f / 127.0f), (16.5f / 127.0f),
+                                     (17.0f / 127.0f),
+                                     (17.5f / 127.0f)) /*Expected Value*/,
+                        15000 /*Precision in ULP*/);
+  }
+  {
+    s::cl_float4 Coord(0.0f, 0.2f, 0.5f, 0.0f); // In-range
+    auto Sampler =
+        s::sampler(s::coordinate_normalization_mode::unnormalized,
+                   s::addressing_mode::clamp, s::filtering_mode::linear);
+    checkReadSampler<7>(host_ptr, Sampler, Coord,
+                        s::cl_float4((0.0f / 127.0f), (0.35f / 127.0f),
+                                     (0.7f / 127.0f),
+                                     (1.05f / 127.0f)) /*Expected Value*/,
+                        15000 /*Precision in ULP*/);
+  }
+
+  // addressing_mode::none
+  {
+    s::cl_float4 Coord(1.0f, 2.0f, 3.0f,
+                       0.0f); // In-range for consistent return value.
+    auto Sampler =
+        s::sampler(s::coordinate_normalization_mode::unnormalized,
+                   s::addressing_mode::none, s::filtering_mode::linear);
+    checkReadSampler<8>(host_ptr, Sampler, Coord,
+                        s::cl_float4((74.0f / 127.0f), (75.0f / 127.0f),
+                                     (76.0f / 127.0f),
+                                     (77.0f / 127.0f)) /*Expected Value*/,
+                        15000 /*Precision in ULP*/);
```

Comment:
`15000 /*Precision in ULP*/` is written about 14 times. Please define it as make constant variable available for all these cases.

Reply:
done.